rocketpy/rocket/aero_surface/fins/free_form_fins.py

```diff
@@ -0,0 +1,379 @@
+        self.cpx = 0
+        self.cpy = 0
+        self.cpz = cpz
+        self.cp = (self.cpx, self.cpy, self.cpz)
+
+    def evaluate_geometrical_parameters(self):  # pylint: disable=too-many-statements
+        """
+        Calculates and saves the fin set's geometrical parameters such as the
+        fin area, aspect ratio, and parameters related to roll movement. This
+        method uses the same calculations to those in OpenRocket for free-form
+        fin shapes.
+
+        Returns
+        -------
+        None
+        """
+        # pylint: disable=invalid-name
+        # pylint: disable=too-many-locals
+        # Calculate the fin area (Af) using the Shoelace theorem (polygon area formula)
+        Af = 0
+        for i in range(len(self.shape_points) - 1):
+            x1, y1 = self.shape_points[i]
+            x2, y2 = self.shape_points[i + 1]
+            Af += (y1 + y2) * (x1 - x2)
+        Af = abs(Af) / 2
+        if Af < 1e-6:
+            raise ValueError("Fin area is too small. Check the shape_points.")
+
+        # Calculate aspect ratio (AR) and lift interference factors
+        AR = 2 * self.span**2 / Af  # Aspect ratio
+        tau = (self.span + self.rocket_radius) / self.rocket_radius
+        lift_interference_factor = 1 + 1 / tau
+
+        # Calculate roll forcing interference factor using OpenRocket's approach
+        roll_forcing_interference_factor = (1 / np.pi**2) * (
+            (np.pi**2 / 4) * ((tau + 1) ** 2 / tau**2)
+            + ((np.pi * (tau**2 + 1) ** 2) / (tau**2 * (tau - 1) ** 2))
+            * np.arcsin((tau**2 - 1) / (tau**2 + 1))
+            - (2 * np.pi * (tau + 1)) / (tau * (tau - 1))
+            + ((tau**2 + 1) ** 2 / (tau**2 * (tau - 1) ** 2))
+            * (np.arcsin((tau**2 - 1) / (tau**2 + 1))) ** 2
+            - (4 * (tau + 1))
+            / (tau * (tau - 1))
+            * np.arcsin((tau**2 - 1) / (tau**2 + 1))
+            + (8 / (tau - 1) ** 2) * np.log((tau**2 + 1) / (2 * tau))
+        )
+
+        # Define number of interpolation points along the span of the fin
+        points_per_line = 40  # Same as OpenRocket
+
+        # Initialize arrays for leading/trailing edge and chord lengths
+        chord_lead = np.ones(points_per_line) * np.inf  # Leading edge x coordinates
+        chord_trail = np.ones(points_per_line) * -np.inf  # Trailing edge x coordinates
+        chord_length = np.zeros(
+            points_per_line
+        )  # Chord length for each spanwise section
+
+        # Discretize fin shape and calculate chord length, leading, and trailing edges
+        for p in range(1, len(self.shape_points)):
+            x1, y1 = self.shape_points[p - 1]
+            x2, y2 = self.shape_points[p]
+
+            # Compute corresponding points along the fin span (clamp to valid range)
+            prev_idx = int(y1 * 1.0001 / self.span * (points_per_line - 1))
+            curr_idx = int(y2 * 1.0001 / self.span * (points_per_line - 1))
```

Comment:
Why `1.0001`?

Reply:
I believe this is something related to numerical issues in Java. This comes directly from Open Rocket.

I removed it because I guess it is not needed for us